flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/allocator/SlotSharingSlotAllocator.java

```diff
@@ -97,6 +104,37 @@ public ParallelismAndResourceAssignments assignResources(
         return new ParallelismAndResourceAssignments(assignedSlots, parallelismPerJobVertex);
     }
 
+    @Override
+    public VertexAssignment determineParallelism(ExecutionGraph executionGraph) {
+        // compute map of SlotSharingGroup groups to ExecutionJobVertex
+        Multimap<SlotSharingGroup, ExecutionJobVertex> slotSharingGroups =
+                ArrayListMultimap.create();
+        executionGraph
+                .getAllExecutionVertices()
+                .forEach(
+                        executionVertex -> {
+                            slotSharingGroups.put(
+                                    executionVertex.getJobVertex().getSlotSharingGroup(),
+                                    executionVertex.getJobVertex());
+                        });
+
+        // get JobVertexID and parallelism of JobVertex with highest parallelism per
+        // SlotSharingGroup
+        Map<JobVertexID, Integer> result = new HashMap<>();
+        for (SlotSharingGroup slotSharingGroup : slotSharingGroups.keySet()) {
+            Optional<ExecutionJobVertex> highestParallelism =
+                    slotSharingGroups.get(slotSharingGroup).stream()
+                            .max(Comparator.comparingInt(ExecutionJobVertex::getParallelism));
+            highestParallelism.ifPresent(
+                    executionJobVertex ->
+                            result.put(
+                                    executionJobVertex.getJobVertexId(),
+                                    executionJobVertex.getParallelism()));
+        }
+
+        return new ExecutionGraphAssignments(result);
+    }
```

Comment:
I don't understand this method. If we are interested in the parallelism per `JobVertexID`, then it should be enough to do:

```
final Map<JobVertexID, Integer> jobVertexParallelisms = executionGraph
                .getAllVertices()
                .values()
                .stream()
                .collect(Collectors.toMap(
                        ExecutionJobVertex::getJobVertexId, ExecutionJobVertex::getParallelism));
```

There is no need to ask the `SlotAllocator` for this.